Multisig/UI/Settings/OwnerKeyManagement/AddKeyAsNewOwner/RemoveOwnerFlow.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  RemoveOwnerFlow.swift
+//  Multisig
+//
+//  Created by Moaaz on 5/11/22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+
+class RemoveOwnerFlow: UIFlow {
+    var factory: RemoveOwnerFlowFactory
+    var safe: Safe
+    var ownerToRemove: Address?
+    var prevOwner: Address?
+    var newConfirmations: Int?
+    var removeOwnerTransactionDetails: SCGModels.TransactionDetails?
+
+    internal init(owner: Address, safe: Safe, factory: RemoveOwnerFlowFactory = .init(), navigationController: UINavigationController, completion: @escaping (_ success: Bool) -> Void) {
+        self.factory = factory
+        self.safe = safe
+        self.ownerToRemove = owner
+        super.init(navigationController: navigationController, completion: completion)
+    }
+
+    override func start() {
+        confirmations()
+    }
+
+    func confirmations() {
+        let confirmationsVC = factory.confirmations(
+            step: 1,
+            maxSteps: 2,
+            safe: safe
+        ) { [unowned self] newConfirmations in
+            self.newConfirmations = newConfirmations
+            review()
+        }
+        show(confirmationsVC)
+    }
+
+    func review() {
+        assert(ownerToRemove != nil)
+        assert(newConfirmations != nil)
+
+        let reviewVC = factory.review(
+            step: 2,
+            maxSteps: 2,
+            safe: safe,
+            owner: ownerToRemove!,
+            prevOwner: prevOwner, newThreshold: newConfirmations!) { [unowned self] txDetails in
+                removeOwnerTransactionDetails = txDetails
+                success()
+            }
+        show(reviewVC)
+    }
+
+    func success() {
```

Comment:
this seems to be repeating in several places, maybe we can generalize the owner change flows if that's feasible and has enough common shared code

Reply:
I think this will require a larger refactor right ?, I would go with making it in another ticket

Reply:
ok